src/service/impl/stop-places/index.ts

```diff
@@ -61,20 +64,50 @@ export default (): IStopPlacesService => {
       if (result.errors) {
         return Result.err(new APIError(result.errors));
       }
-      const uniqueStopPlaces = result.data.stopPlace?.quays
+      if (!result.data.stopPlace?.quays) {
+        return Result.ok([]);
+      }
+      const journeyPatternsWithMatchingAuthority = result.data.stopPlace.quays
         ?.flatMap((quay) => quay.journeyPatterns)
```

Comment:
Don't need the `?` here after you added the guard above. Then maybe you can remove some more of the undefined states below?